bundles/org.openhab.binding.somneo/src/main/java/org/openhab/binding/somneo/internal/SomneoHandler.java

```diff
@@ -0,0 +1,542 @@
+            return;
+        }
+
+        pollingJob.cancel(true);
+        this.pollingJob = null;
+        logger.debug("HTTP polling stopped.");
+    }
+
+    private void poll() {
+        final SomneoHttpConnector connector = this.connector;
+        if (connector == null) {
+            return;
+        }
+
+        try {
+            final SensorData sensorData = connector.fetchSensorData();
+            updateState(CHANNEL_SENSOR_HUMIDITY, sensorData.getCurrentHumidity());
+            updateState(CHANNEL_SENSOR_ILLUMINANCE, sensorData.getCurrentIlluminance());
+            updateState(CHANNEL_SENSOR_NOISE, sensorData.getCurrentNoise());
+            updateState(CHANNEL_SENSOR_TEMPERATURE, sensorData.getCurrentTemperature());
+
+            final LightData lightData = connector.fetchLightData();
+            updateState(CHANNEL_LIGHT_MAIN, lightData.getMainLightState());
+            updateState(CHANNEL_LIGHT_NIGHT, lightData.getNightLightState());
+            lastLightBrightness = lightData.getMainLightLevel();
+
+            final SunsetData sunsetData = connector.fetchSunsetData();
+            updateState(CHANNEL_SUNSET_SWITCH, sunsetData.getSwitchState());
+            updateState(CHANNEL_SUNSET_LIGHT_INTENSITY, sunsetData.getLightIntensity());
+            updateState(CHANNEL_SUNSET_DURATION, sunsetData.getDurationInMin());
+            updateState(CHANNEL_SUNSET_COLOR_SCHEMA, sunsetData.getColorSchema());
+            updateState(CHANNEL_SUNSET_AMBIENT_NOISE, sunsetData.getAmbientNoise());
+            updateState(CHANNEL_SUNSET_VOLUME, sunsetData.getSoundVolume());
+
+            final RelaxData relaxData = connector.fetchRelaxData();
+            updateState(CHANNEL_RELAX_SWITCH, relaxData.getSwitchState());
+            updateState(CHANNEL_RELAX_BREATHING_RATE, relaxData.getBreathingRate());
+            updateState(CHANNEL_RELAX_DURATION, relaxData.getDurationInMin());
+            updateState(CHANNEL_RELAX_GUIDANCE_TYPE, relaxData.getGuidanceType());
+            updateState(CHANNEL_RELAX_LIGHT_INTENSITY, relaxData.getLightIntensity());
+            updateState(CHANNEL_RELAX_VOLUME, relaxData.getSoundVolume());
+
+            final AudioData audioData = connector.fetchAudioData();
+            updateState(CHANNEL_AUDIO_RADIO, audioData.getRadioState());
+            updateState(CHANNEL_AUDIO_AUX, audioData.getAuxState());
+            updateState(CHANNEL_AUDIO_VOLUME, audioData.getVolumeState());
+            updateState(CHANNEL_AUDIO_PRESET, audioData.getPresetState());
+
+            updateFrequency();
+
+            updateRemainingTimer();
+
+            updateStatus(ThingStatus.ONLINE);
+        } catch (TimeoutException e) {
+            logger.warn("Timeout: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (InterruptedException e) {
+            logger.debug("Polling data interrupted");
+        } catch (Exception e) {
```

Comment:
Again, I don't think all exceptions should be caught here. Which exception type is thrown having `EOFException` as cause?